packages/ec-core/src/operator/selector/lexicase.rs

```diff
@@ -54,22 +74,35 @@ where
         case_indices.shuffle(rng);
 
         let mut candidates: Vec<_> = population.into_iter().collect();
-
         let mut winners = Vec::with_capacity(candidates.len());
+
         for test_case_index in case_indices {
-            assert!(
-                candidates.is_empty().not(),
-                "The set of lexicase candidates shouldn't be empty"
-            );
-            if candidates.len() == 1 {
+            let (&initial_winner, remaining) = candidates.split_first().ok_or(EmptyPopulation)?;
+
+            if remaining.is_empty() {
                 break;
             }
+
             winners.clear();
-            winners.push(candidates[0]);
-            let mut current_best_result = &winners[0].test_results().results[test_case_index];
+            winners.push(initial_winner);
+
+            let mut current_best_result = initial_winner
+                .test_results()
+                .results
+                .get(test_case_index)
+                .ok_or(LexicaseError::MissingTestCase {
```

Comment:
Should this be `ok_or_else()`? I realize that there's not any "interesting" computation here, but the `MissingTestCase` struct is getting rebuilt every time through the loop, and will almost never actually get used.

Reply:
I went with `ok_or_else` at first as well but clippy wanted it this way 🤷  I'd leave it like this until we see that it makes a measurable performance impact